dev/tox.ini

```diff
@@ -20,5 +20,16 @@ exclude=python/pyspark/cloudpickle/*.py,shared.py,python/docs/source/conf.py,wor
 
 [flake8]
 select = E901,E999,F821,F822,F823,F401,F405
-exclude = python/pyspark/cloudpickle/*.py,shared.py,python/docs/source/conf.py,work/*/*.py,python/.eggs/*,dist/*,.git/*
+exclude = python/pyspark/cloudpickle/*.py,shared.py*,python/docs/source/conf.py,work/*/*.py,python/.eggs/*,dist/*,.git/*,python/out
 max-line-length = 100
+per-file-ignores =
+    python/pyspark/sql/tests/test_arrow.py: F405
+    python/pyspark/sql/tests/test_dataframe.py: F405
+    python/pyspark/sql/tests/test_pandas_udf_scalar.py: F405
+    python/pyspark/sql/tests/test_udf.py: F405
+    python/pyspark/testing/streamingutils.py: F821
+    python/pyspark/testing/mlutils.py: F821
```

Comment:
Looks like these are mostly because flake8 doesn't understand `# type: ignore[<code>]` - finds `<code>` is a proper instance to check if it's being imported or not(?). If that's the case, let's just use `# type: ignore`. I think this is the easiest way.

Reply:
Sounds reasonable, though I'd prefer to use mixed strategy:

- Less specific `type: ignore` for non-pyi files.
- Ignore selected `pyi`  files in flake8.

This way we can get the most coverage.